webserver-ubuntu.sh

```diff
@@ -16,20 +14,37 @@ then
   echo "Git is installed"
 else
   echo "Installing Git"
-  sudo apt install git
+  apt install -y git
 fi
 
+
 echo "Checking if Multipass is installed" 
 if ( multipass version ) 
 then
   echo "Multipass is installed" 
 else 
   echo "Installing Multipass" 
-  git clone https://aur.archlinux.org/snapd.git
-  cd snapd
-  makepkg -si
-  cd ..
-  sudo systemctl enable --now snapd.socket
-  sudo ln -s /var/lib/snapd /snap
-  sudo snap install multipass
+  # the following should not be necessary on recent versions of Ubuntu
+  # checking for snapd anyway
+  if ( which snap )
```

Comment:
Let's check for the thing we want, for example "multipass should be installed". We don't want snapd, it's just how we're getting multipass.

Looking at this code, it looks like what we want is "snapd" and then we install multipass without any conditional on it which is not what we want.

Reply:
The check for multipass is before the check for snap. The latter is a nested conditional, only checked if the check for multipass fails as I don't need to check for snap/snapd if multipass is present.

Reply:
Removed "exec bash" as I had hoped that by restarting the shell that it might also set up snapd's env variables/path per the instructions here: https://snapcraft.io/docs/installing-snap-on-debian

Unfortunately this is not the case and this script currently works on debian but multipass is not on the system path so it's not possible to launch multipass right after installation. Per the snapd installation instructions above, restarting the VM did result in the state that we want: multipass is available on the path/ready to launch a new vm

Tested on a debian VM on virtualbox because even the older versions of Ubuntu available on multipass have snapd pre-installed and I wanted to test a version of linux that uses the apt package manager that would not have snapd installed.